File: shoring/flywheel/src/main/java/com/panopset/flywheel/TokenVariableFactory.java
package com.panopset.flywheel;

import java.util.Map;
import java.util.StringTokenizer;
import com.panopset.compat.Stringop;

public class TokenVariableFactory {

  public void addTokensToMap(Map<String, String> map, String line, String tokens) {
    if (Stringop.isPopulated(tokens)) {
      int i = 0;
      StringTokenizer st = new StringTokenizer(line, tokens);
      while (st.hasMoreTokens()) {
        map.put(String.format("t%d", i++), st.nextToken());
      }
    }
  }
}

File: code/shoring/flywheel/src/main/kotlin/com/panopset/flywheel/TokenVariableFactory.kt
package com.panopset.flywheel

import com.panopset.compat.Stringop
import java.io.StringWriter
import java.util.*

class TokenVariableFactory {
    fun addTokensToMap(map: MutableMap<String, String>, line: String, tokens: String) {
        if (Stringop.isPopulated(tokens)) {
            var i = 0
            val st = StringTokenizer(line, tokens)
            while (st.hasMoreTokens()) {
                map[String.format("t%d", i++)] = st.nextToken()
            }
        }
    }

    fun addTokensToMap(map: MutableMap<String, String>, line: String, tokens: String, template: Template) {
        if (Stringop.isPopulated(tokens)) {
            var i = 0
            val st = StringTokenizer(line, tokens)
            while (st.hasMoreTokens()) {
                map[String.format("t%d", i++)] = processValue(template, st.nextToken())
            }
        }
    }

    /**
     * TODO: Duplicate in CommandList
     * Process a list value, resolving any template commands within the line.
     *
     * @param str
     * String to be processed.
     * @return processed value.
     */
    private fun processValue(template: Template, str: String): String {
        val sw = StringWriter()
        Template(template.flywheel, TemplateArray(arrayOf(str)), LFR_FLATTEN).exec(sw)
        return sw.toString()
    }
}
